### PR TITLE
Add warning about using tiflash in read write txn

### PR DESCRIPTION
**Pull request by zanmato1984**
### What is changed, added or deleted? (Required)

Add a warning about using tiflash in read write transaction.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: tiflash/use-tiflash.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ TiFlash 部署完成后并不会自动同步数据，而需要手动指定需要
 - [使用 TiDB 读取 TiFlash](#使用-tidb-读取-tiflash)
 - [使用 TiSpark 读取 TiFlash](#使用-tispark-读取-tiflash)
 
+> **注意：**
+>
+> 如果在含有写的事务中（例如事务中包含 `SELECT ... FOR UPDATE` 接 `UPDATE ...`）读取 TiFlash（见 [使用 TiDB 读取 TiFlash](#使用-tidb-读取-tiflash) ），目前行为是未定义，后续版本中会取消这个限制。
```
Comment (yikeke) [inline tiflash/use-tiflash.md:18]:
```suggestion
> 如果在含有写的事务中（例如事务中包含 `SELECT ... FOR UPDATE` 且后接 `UPDATE ...`）读取 TiFlash（见 [使用 TiDB 读取 TiFlash](#使用-tidb-读取-tiflash) ），目前行为是未定义，后续版本中会取消这个限制。
```
这个意思？

Comment (zanmato1984) [reply tiflash/use-tiflash.md:18]: 恩，是你修改后的例子。Suggestion accepted.

Comment (yikeke) [reply tiflash/use-tiflash.md]: 可以私戳马老师或者李坤老师点个赞哈 @zanmato1984

File: tiflash/use-tiflash.md [hunk 2]
```diff
@@ -13,6 +13,10 @@ TiFlash 部署完成后并不会自动同步数据，而需要手动指定需要
 - [使用 TiDB 读取 TiFlash](#使用-tidb-读取-tiflash)
 - [使用 TiSpark 读取 TiFlash](#使用-tispark-读取-tiflash)
 
+> **注意：**
+>
+> 如果在含有写的事务中（例如事务中包含 `SELECT ... FOR UPDATE` 且后接 `UPDATE ...`）读取 TiFlash（见 [使用 TiDB 读取 TiFlash](#使用-tidb-读取-tiflash) ），目前行为是未定义，后续版本中会取消这个限制。
```
Comment (yikeke) [inline tiflash/use-tiflash.md:18]:
```suggestion
> 如果在含有写的事务中（例如事务中包含 `SELECT ... FOR UPDATE` 且后接 `UPDATE ...`）读取 TiFlash（见[使用 TiDB 读取 TiFlash](#使用-tidb-读取-tiflash)），目前行为是未定义，后续版本中会取消这个限制。
```

File: tiflash/use-tiflash.md [hunk 3]
```diff
@@ -13,6 +13,10 @@ TiFlash 部署完成后并不会自动同步数据，而需要手动指定需要
 - [使用 TiDB 读取 TiFlash](#使用-tidb-读取-tiflash)
 - [使用 TiSpark 读取 TiFlash](#使用-tispark-读取-tiflash)
 
+> **注意：**
+>
+> 如果在含有写的事务中（例如事务中包含 `SELECT ... FOR UPDATE` 且后接 `UPDATE ...`）读取 TiFlash（见[使用 TiDB 读取 TiFlash](#使用-tidb-读取-tiflash)），目前行为是未定义，后续版本中会取消这个限制。
```
Comment (yikeke) [inline tiflash/use-tiflash.md:18]:
```suggestion
> 如果在含有写的事务中（比如事务中包含 `SELECT ... FOR UPDATE` 且后接 `UPDATE ...`）读取 TiFlash（见[使用 TiDB 读取 TiFlash](#使用-tidb-读取-tiflash)），目前行为是未定义，后续版本中会取消这个限制。
```

**Pull request merged**